src/osd/OSD.cc

```diff
@@ -8269,11 +8269,13 @@ void OSD::do_recovery(PG *pg, ThreadPool::TPHandle &handle)
 {
   if (g_conf->osd_recovery_sleep > 0) {
     handle.suspend_tp_timeout();
+    pg->unlock();
```

Comment:
This is wrong.  In this version of the code, we haven't locked the PG yet here, which is why jewel doesn't already unlock the pg here.  This change was only needed in master due to a change in the way the queuing worked.  Honestly, this should be removed altogether in master and is of questionable value in jewel.